src/libertem/udf/base.py

```diff
@@ -145,6 +148,19 @@ def tiling_scheme(self) -> Optional[TilingScheme]:
         """
         return self._tiling_scheme
 
+    @property
+    def tiling_scheme_idx(self) -> int:
+        return self._tiling_index
+
+    @tiling_scheme_idx.setter
+    def tiling_scheme_idx(self, new_idx: int) -> None:
+        self._tiling_index = new_idx
+
+    @property
+    def sig_slice(self) -> Slice:
+        assert self._tiling_scheme is not None
+        return self._tiling_scheme[self._tiling_index]
```

Comment:
Ok, so the point is that the sig slice is reusable for tiles with the same tiling index, while the tile slice changes with each tile, right?

Reply:
Indeed - churning through additional objects with `tile_slice.discard_nav()` or similar does create a measurable overhead for throughput-limited workloads like `ApplyMasksUDF`